src/request.rs

```diff
@@ -119,10 +119,10 @@ impl SecretType for SecretKey {
     }
 
     fn policies(&self) -> Vec<policy::Policy> {
-        if let Some(p) = db::get_secret_policy(&self.request.id) {
-            return vec![p];
+        match db::get_secret_policy(&self.request.id) {
+            Some(p) => return vec![p],
+            None => return vec![],
```

Comment:
I think you can remove `return` from both lines, since the `match` is the last (only) statement in the function.

Reply:
Done